src/web/lib/storage.js

```diff
@@ -17,16 +17,26 @@ const generateThemeKey = () =>
   `${Date.now()}-${Math.floor(Math.random() * 1000)}`;
 
 function putTheme(key, theme) {
-  log("putTheme", key, theme);
-  const storageKey = themeStorageKey(key);
-  localStorage.setItem(
-    storageKey,
-    JSON.stringify({
-      theme,
-      modified: Date.now()
-    })
-  );
-  notifySelfForStorage(storageKey);
+  let isThemeDuplicate = checkDuplicateTheme(theme);
+  if (!isThemeDuplicate) {
+    log("putTheme", key, theme);
+    const storageKey = themeStorageKey(key);
+    localStorage.setItem(
+      storageKey,
+      JSON.stringify({
+        theme,
+        modified: Date.now()
+      })
+    );
+    notifySelfForStorage(storageKey);
+  }
```

Comment:
Semi-related, but do we need an `else` statement?
Re-reading @johngruen's https://github.com/mozilla/FirefoxColor/issues/166#issuecomment-373050617, it sounded like we may want to pop the newly saved duplicate theme to the front of the list if it was a duplicate. Not sure if the code is currently doing that and I'm just not reading it correctly, or if we need an `else` where we update the `modified` date to `Date.now()` and do whatever animations we wanted.

Reply:
Can we merge this code and open a new issue to include animation effect?

Reply:
Hi @meandavejustice that sounds good.
While testing the changes after resolving the conflicts, I found that 'Save' button is disabled if there is no change from 'currentSaved' theme.
One case where this functionality (of checking duplicate saved themes) would be useful is when a user presses 'undo/redo' button and tries to save the same theme again.
So, would you still recommend to add this functionality?
Please let me know if I'm missing something here.

Reply:
@pdehaan @meandavejustice Could you please review if we still need to fix this issue?
If not, we can close this PR.